File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        if 0 in heights and len(heights)==2:
            return 0
        left = 0
        right= len(heights)-1
        maximumAmountOfWater=1
        while left < right:
            w = abs(right - left)
            h = min(heights[right],heights[left])
            maximumAmountOfWater = max(h*w,maximumAmountOfWater)
            if heights[left] < heights[right]:
                left +=1
            else: right -=1
        return maximumAmountOfWater

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right=height.size()-1;
        int maximumAmountOfWater = 0;
        int h=0, w=0;
        while(left < right){
            w = abs(left - right);
            h = min(height[left], height[right]);

            maximumAmountOfWater = max(maximumAmountOfWater, h*w);

            while(left < right && height[left] <=h) left++;
            while(left < right && height[right] <=h) right--;
        }
        return maximumAmountOfWater;
    }
};